care/facility/api/viewsets/patient.py

```diff
@@ -279,6 +283,52 @@ def filter_by_diagnoses(self, queryset, name, value):
             )
         return queryset.filter(filter_q)
 
+    last_consultation__consent_types = MultiSelectFilter(
+        method="filter_by_has_consents"
+    )
+
+    def filter_by_has_consents(self, queryset, name, value: str):
+
+        if not value:
+            return queryset
+
+        values = value.split(",")
+
+        filter_q = Q()
+
+        consultation_consent_files = (
+            FileUpload.objects.filter(
+                file_type=FileUpload.FileType.CONSENT_RECORD,
+                is_archived=False,
+            )
+            .order_by("associating_id")
+            .values_list("associating_id", flat=True)
+        )
+
+        consultations = PatientConsent.objects.filter(
+            external_id__in=[x for x in consultation_consent_files],
```

Comment:
```suggestion
            external_id__in=consultation_consent_files,
```
consultation_consent_files is already a list

Reply:
this is for typecasting, without it you get errors.

Reply:
Can you try to fix the queryset instead, it would be better if we can execute it without converting it to python

Reply:
I dont quite understand what I should do to fix it, but even if we keep it this way, two queries are going to be made any ways, so is there any performance implications for this?